core/commands/storage/upload/upload/dc_repair_router.go

```diff
@@ -0,0 +1,265 @@
+		ShortDescription: `
+Guard broadcasts repair request to potential nodes, after negotiation, nodes prepare the contracts for 
+such repair job, sign and send to the guard for confirmation `,
+	},
+	Subcommands: map[string]*cmds.Command{
+		"request":  hostRepairRequestCmd,
+		"response": hostRepairResponseCmd,
+	},
+}
+
+type RepairContractParams struct {
+	FileHash             string
+	FileSize             int64
+	RepairPid            string
+	RepairSignTime       time.Time
+	LostShardHashes      []string
+	RepairContractId     string
+	RepairRewardAmount   int64
+	DownloadContractId   string
+	DownloadRewardAmount int64
+}
+
+var hostRepairRequestCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Negotiate with hosts for repair jobs.",
+		ShortDescription: `
+This command sends request to mining host to negotiate the repair works.`,
+	},
+	Arguments: append([]cmds.Argument{
+		cmds.StringArg("peer-ids", true, false, "Host Peer IDs to send repair requests."),
+	}, hostRepairResponseCmd.Arguments...), // append pass-through arguments
+	RunTimeout: 20 * time.Second,
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		fileHash := req.Arguments[1]
+		repairId := req.Arguments[6]
+		repairContractId := req.Arguments[8]
+		lostShardHashes := strings.Split(req.Arguments[2], ",")
+		fileSize, err := strconv.ParseInt(req.Arguments[3], 10, 64)
+		if err != nil {
+			return err
+		}
+		downloadRewardAmount, err := strconv.ParseInt(req.Arguments[4], 10, 64)
+		if err != nil {
+			return err
+		}
+		repairRewardAmount, err := strconv.ParseInt(req.Arguments[5], 10, 64)
+		if err != nil {
+			return err
+		}
+		err = emptyCheck(fileHash, lostShardHashes, repairId, repairContractId, fileSize, downloadRewardAmount, repairRewardAmount)
+		if err != nil {
+			return nil
+		}
+		cfg, err := cmdenv.GetConfig(env)
+		if err != nil {
+			return err
+		}
+		if !cfg.Experimental.StorageClientEnabled {
+			return fmt.Errorf("storage client api not enabled")
+		}
+		if !cfg.Experimental.HostRepairEnabled {
+			return fmt.Errorf("host repair api not enabled")
+		}
+		n, err := cmdenv.GetNode(env)
+		if err != nil {
+			return err
+		}
+		api, err := cmdenv.GetApi(env, req)
+		if err != nil {
+			return err
+		}
+		peerIds := strings.Split(req.Arguments[0], ",")
+		reapirIds := make([]string, len(peerIds))
+		var wg sync.WaitGroup
+		wg.Add(len(peerIds))
+		for index, peerId := range peerIds {
+			go func() {
+				err = func() error {
+					pi, err := remote.FindPeer(req.Context, n, peerId)
+					if err != nil {
+						return err
+					}
+					_, err = remote.P2PCallStrings(req.Context, n, api, pi.ID, "/storage/upload/dcrepairrouting/response",
+						req.Arguments[1:]...)
+					if err != nil {
+						return err
+					}
+					reapirIds[index] = peerId
+					return nil
+				}()
+				if err != nil {
+					log.Debug("P2P call error for peerId %s", peerId)
```

Comment:
it is better to record the exact error here in the log, use something like log.Error("P2P call error for peerId", zap.error(err), zap.string("peerId",peerId))

Reply:
updated